From ed6dc31ed1092baf42f99278ef9c5c297805c3d0 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 24 Apr 2014 08:58:20 +0200
Subject: [PATCH 2/2] Misc header fixes

src/bindresvport.c: IPPORT_RESERVED needs netdb.h

Misc fixes for building on musl libc
---
 src/bindresvport.c | 1 +
 tirpc/netconfig.h  | 1 +
 tirpc/rpc/rpcent.h | 2 +-
 tirpc/rpc/types.h  | 1 +
 4 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/bindresvport.c b/src/bindresvport.c
index d6d9c14..5a7a1a9 100644
--- a/src/bindresvport.c
+++ b/src/bindresvport.c
@@ -37,6 +37,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 
+#include <netdb.h>
 #include <netinet/in.h>
 
 #include <errno.h>
diff --git a/tirpc/netconfig.h b/tirpc/netconfig.h
index 7d6c2bc..2922e6f 100644
--- a/tirpc/netconfig.h
+++ b/tirpc/netconfig.h
@@ -2,6 +2,7 @@
 #define _NETCONFIG_H_
 
 #include <features.h>
+#include <sys/cdefs.h>
 
 #define NETCONFIG "/etc/netconfig"
 #define NETPATH	  "NETPATH"
diff --git a/tirpc/rpc/rpcent.h b/tirpc/rpc/rpcent.h
index c865e51..a36cf91 100644
--- a/tirpc/rpc/rpcent.h
+++ b/tirpc/rpc/rpcent.h
@@ -47,7 +47,7 @@
 __BEGIN_DECLS
 
 /* These are defined in /usr/include/rpc/netdb.h */
-#if 0
+#if !defined(__GLIBC__)
 struct rpcent {
 	char	*r_name;	/* name of server for this rpc program */
 	char	**r_aliases;	/* alias list */
diff --git a/tirpc/rpc/types.h b/tirpc/rpc/types.h
index 52c30a2..d967271 100644
--- a/tirpc/rpc/types.h
+++ b/tirpc/rpc/types.h
@@ -39,6 +39,7 @@
 #define _TIRPC_TYPES_H
 
 #include <sys/types.h>
+#include <sys/cdefs.h>
 
 typedef int32_t bool_t;
 typedef int32_t enum_t;
-- 
1.9.2

